src/agentscope/agent/_react_agent.py

```diff
@@ -225,9 +225,14 @@ def __init__(
         self.plan_notebook = None
         if plan_notebook:
             self.plan_notebook = plan_notebook
+            # When enable_meta_tool is True, the plan tool group activation is
+            # controlled by agent.
+            # Otherwise, the plan tool group is always active.
+            tool_group_active = not enable_meta_tool
             self.toolkit.create_tool_group(
```

Comment:
Thank you for your contribution. However, there appears to be a misunderstanding. When enable_meta_tool is set to False, tools should be added directly via register_tool_function (registering them into the "default" group), rather than creating a separate tool group.

Reply:
Done.

Register plan tools in basic group if enable_meta_tool is not True